youtube_dl/extractor/picarto.py

```diff
@@ -0,0 +1,80 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+import time
+
+from .common import InfoExtractor
+from ..utils import ExtractorError
+from ..compat import compat_str
+
+
+class PicartoIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?picarto\.tv/(?P<id>[a-zA-Z0-9]+)'
+    _TEST = {
+        'url': 'https://picarto.tv/setz',
+        'info_dict': {
+            'id': 'Setz',
+            'ext': 'mp4',
+            'title': 're:^Setz [0-9]{4}-[0-9]{2}-[0-9]{2} [0-9]{2}:[0-9]{2}$',  # match self._live_title
+            'timestamp': int,
+            'is_live': True
+        },
+        'params': {
+            # Livestream
+            'skip_download': True
+        }
+    }
+    _PLACESTREAM_REGEX = r'''(?x)
+                             placeStream\(
+                                "(?P<channel>[a-zA-Z0-9]+)",
+                                (?P<player_id>\d+),
+                                (?P<product>\d+),
```

Comment:
I was being verbose as the placeStream may change in the future, and some of them can be used to fetch a thumbnail or viewer counts, which I didn't consider crucial for first implementation.